src/components/domain/RiskImminentEvents/Pdc/index.tsx

```diff
@@ -59,28 +59,76 @@ function Pdc(props: Props) {
         },
     });
 
-    const { trigger: getFootprint } = useRiskLazyRequest<'/api/v1/pdc/{id}/exposure/', {
-        successCallback: FootprintCallback,
+    const {
+        response: exposureResponse,
+        pending: exposureResponsePending,
+        trigger: getFootprint,
+    } = useRiskLazyRequest<'/api/v1/pdc/{id}/exposure/', {
         eventId: number | string,
     }>({
         apiType: 'risk',
         url: '/api/v1/pdc/{id}/exposure/',
         pathVariables: ({ eventId }) => ({ id: Number(eventId) }),
-        onSuccess: (response, { successCallback }) => {
+    });
+
+    const pointFeatureSelector = useCallback(
+        (event: EventItem): EventPointFeature | undefined => {
+            const {
+                id,
+                latitude,
+                longitude,
+                hazard_type,
+            } = event;
+
+            if (isNotDefined(latitude)
+                || isNotDefined(longitude)
+                || isNotDefined(hazard_type)
+                || hazard_type === ''
+            ) {
+                return undefined;
+            }
+
+            return {
+                type: 'Feature',
+                geometry: {
+                    type: 'Point',
+                    coordinates: [longitude, latitude],
+                },
+                properties: {
+                    id,
+                    hazard_type,
+                },
+            };
+        },
+        [],
+    );
+
+    const footprintSelector = useCallback(
+        (exposure: RiskApiResponse<'/api/v1/pdc/{id}/exposure/'> | undefined) => {
+            if (isNotDefined(exposure)) {
+                return undefined;
+            }
+
             const {
                 footprint_geojson,
                 storm_position_geojson,
-            } = response;
+            } = exposure;
 
             if (isNotDefined(footprint_geojson) && isNotDefined(storm_position_geojson)) {
-                return;
+                return undefined;
             }
 
             const footprint = isValidFeature(footprint_geojson) ? footprint_geojson : undefined;
             // FIXME: typings should be fixed in the server
             const stormPositions = (storm_position_geojson as unknown as unknown[] | undefined)
                 ?.filter(isValidPointFeature);
 
+            // forecast_date_time : "2023 SEP 04, 00:00Z"
+            // severity : "WARNING"
+            // storm_name : "HAIKUI"
+            // track_heading : "WNW"
+            // wind_speed_mph : 75
+
```

Comment:
Any reason we need this?

Reply:
Yep, these are the properties in geojson for the reference